newbank/server/NewBankClientHandler.java

```diff
@@ -0,0 +1,64 @@
+package newbank.server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+public class NewBankClientHandler extends Thread{
+	
+	private newbank.server.NewBank bank;
+	private BufferedReader in;
+	private PrintWriter out;
+	
+	
+	public NewBankClientHandler(Socket s) throws IOException {
+		bank = newbank.server.NewBank.getBank();
+		in = new BufferedReader(new InputStreamReader(s.getInputStream()));
+		out = new PrintWriter(s.getOutputStream(), true);
+	}
+	
+	public void run() {
+		// keep getting requests from the client and processing them
+		try {
+			newbank.server.CustomerID customer = null;
+			int times = 0;
+			while (customer == null){
+				if (times>0){
+					out.println("Log In Failed. Please re-enter your details");
+				}
+				// ask for user name
+				out.println("Enter Username");
+				String userName = in.readLine();
```

Comment:
I think potentially some scope for some abstraction of common methods here, potentially a function that prints a line and then returns the string value,

For example, 

	public String getUserInput(String message) throws IOException {
		this.out.println(message);
		return this.in.readLine();
	}

Reply:
Done, you're right, it looks simpler that way.